lms/djangoapps/courseware/views.py

```diff
@@ -711,6 +711,21 @@ def course_info(request, course_id):
         if settings.FEATURES.get('ENABLE_MKTG_SITE'):
             url_to_enroll = marketing_link('COURSES')
 
+        # only allow course sneak peek if
+        # 1) within enrollment period
+        # 2) course specifies it's okay
+        # 3) request.user is not a registered user.
+        sneakpeek_allowed = (has_access(request.user, 'within_enrollment_period', course) and
+                             CoursePreference.course_allows_nonregistered_access(course_key) and
+                             not UserProfile.has_registered(request.user))
+
+        registered = registered_for_course(course, request.user)
+        regularly_registered = (
```

Comment:
It would be nice if we could avoid the additional DB lookups here, since these are on every `/info` page load and that's decently trafficked, being a public page.

At the very least, I think we can avoid the `registered_for_course` and `UserProfile.has_registered` checks if `sneakpeek_allowed` is `False`, since we don't care if they were sneakpeek elsewhere if it's not enabled for this course.